common/reactive/src/main/java/io/helidon/common/reactive/Mono.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.common.reactive;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import io.helidon.common.reactive.Flow.Publisher;
+import io.helidon.common.reactive.Flow.Subscriber;
+
+/**
+ * Single item publisher.
+ * @param <T> published type
+ */
+public interface Mono<T> extends Publisher<T> {
+
+    /**
+     * Retrieve the value of this {@link Mono} instance in a blocking manner.
+     *
+     * @return value
+     * @throws IllegalStateException if the Mono wraps an error or if
+     * interrupted
+     */
+    default T block() {
+        MonoBlockingSubscriber<T> subscriber = new MonoBlockingSubscriber<>();
+        this.subscribe(subscriber);
+        return subscriber.blockingGet();
+    }
+
+    /**
+     * Retrieve the value of this {@link Mono} instance in a blocking manner.
+     *
+     * @param timeout timeout value
+     * @return value
+     * @throws IllegalStateException if the Mono wraps an error, or the
+     * timeout is reached or if interrupted
+     */
+    default T block(Duration timeout) {
+        MonoBlockingSubscriber<T> subscriber = new MonoBlockingSubscriber<>();
+        this.subscribe(subscriber);
+        return subscriber.blockingGet(timeout.toMillis(),
+                TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Map this {@link Mono} instance to a new {@link Mono} of another type
+     * using the given {@link MonoMapper}.
+     * @param <U> mapped item type
+     * @param mapper mapper
+     * @return Mono
+     */
+    default <U> Mono<U> map(MonoMapper<T, U> mapper) {
```

Comment:
I've made all the processors package private and exposed interfaces instead.
I've created my own Mapper interface for now since the common-mapper is not available yet ; we need to get a PR going for that separately out of the dbclient branch.